flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -567,28 +565,14 @@ private void addEdgeInternal(Integer upStreamVertexID,
 				shuffleMode = ShuffleMode.UNDEFINED;
 			}
 
-			StreamEdge edge = new StreamEdge(upstreamNode, downstreamNode, typeNumber, outputNames, partitioner, outputTag, shuffleMode);
+			StreamEdge edge = new StreamEdge(upstreamNode, downstreamNode, typeNumber,
+				partitioner, outputTag, shuffleMode);
 
 			getStreamNode(edge.getSourceId()).addOutEdge(edge);
 			getStreamNode(edge.getTargetId()).addInEdge(edge);
 		}
 	}
 
-	public <T> void addOutputSelector(Integer vertexID, OutputSelector<T> outputSelector) {
```

Comment:
It seems we should also remove `addVirtualSelectNode` method, the related `virtualSelectNodes` map and its occurrence ?